extensions/subscriptions/src/Api/UserResourceFields.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Subscriptions\Api;
+
+use Flarum\Api\Context;
+use Flarum\Api\Schema;
+use Flarum\Discussion\Discussion;
+
+class UserResourceFields
+{
+    public function __invoke(): array
+    {
+        return [
+            Schema\Str::make('subscription')
+                ->writable(fn (Discussion $discussion, Context $context) => $context->updating() && ! $context->getActor()->isGuest())
```

Comment:
> Should any fields be writable by default for guests?

the checking here is actually unnecessary since the user update endpoint is for registered users only. But yes if an endpoint is open to guests but a specific field shouldn't be then writability has to be scoped.

there probably won't be many instances of guest write access to an endpoint. Like the User::Create is a unique case for registration.